src/main/kotlin/ecommerce/service/ProductService.kt

```diff
@@ -0,0 +1,25 @@
+package ecommerce.service
+
+import ecommerce.model.Product
+import ecommerce.repository.ProductRepository
+import org.springframework.stereotype.Service
+
+@Service
+class ProductService(private val productRepository: ProductRepository) {
```

Comment:
You used `Constructor Injection` for DI 👍 
Have you heard of `Field Injection` and `Setter Injection` as well? What are the differences between them?